File: andriodGL/app/src/main/java/z3d/display/particle/ctrl/AxisRotaion.java
package z3d.display.particle.ctrl;

import java.util.List;

import z3d.base.Scene_data;
import z3d.vo.Vector3D;

public class AxisRotaion extends BaseAnim {
    public Vector3D axis;
    public Vector3D axisPos;

    public void dataByte(List va, List arr) {
//        this.beginTime =(float) arr.get(0);
//        float indx001=(float)  arr.get(1);
//        if (indx001== -1) {
//            this.lastTime = Scene_data.MAX_NUMBER;
//        } else {
//            this.lastTime = indx001;
//        }
        super.dataByte(va,arr);
        this.axis =(Vector3D) arr.get(2);
        this.axisPos =(Vector3D)arr.get(3);
        this.speed = (float)arr.get(4) * 0.1f;
        this.aSpeed =(float)arr.get(5) * 0.1f;

    }
}

File: iphone/MetalTriangleDemo/MetalTriangleDemo/pan3d/display/particle/ctrl/AxisRotaion.m
//
//  AxisRotaion.m
//  iosgl
//
//  Created by zhao on 31/3/2020.
//  Copyright © 2020 zhao. All rights reserved.
//

#import "AxisRotaion.h"
#import "Scene_data.h"

@implementation AxisRotaion
-(void)dataByte:(NSArray<NSNumber*>*)va arr:(NSArray<NSNumber*>*)arr;
{
    AxisRotaion* this=self;
    this.beginTime = [arr[0]floatValue];
          if ([arr[1]intValue] == -1) {
              this.lastTime =  [Scene_data default] .MAX_NUMBER;
          } else {
              this.lastTime = [arr[1] floatValue];
          }
    this.axis = arr[2];
    this.axisPos = arr[3];
          this.speed = [arr[4]floatValue] * 0.1;
          this.aSpeed = [arr[5]floatValue] * 0.1;
}
@end
